unlock-app/src/utils/checkoutLockUtils.ts

```diff
@@ -145,29 +148,61 @@ export const inClaimDisallowList = (address: string) => {
   return claimDisallowList.indexOf(address) > -1
 }
 
-/**
- * Helper function that returns a valid referrer address
- * @param recipient
- * @param paywallConfig
- * @param lockAddress
- * @returns
- */
-export const getReferrer = (
-  recipient: string,
+export const shouldReferrerResolveForENS = (
   paywallConfig?: PaywallConfigType,
   lockAddress?: string
-): string => {
+) => {
   if (paywallConfig) {
     if (
       lockAddress &&
       paywallConfig.locks[lockAddress] &&
       isAccount(paywallConfig.locks[lockAddress].referrer)
     ) {
-      return paywallConfig.locks[lockAddress].referrer!
+      return false
     }
-    if (paywallConfig.referrer && isAccount(paywallConfig.referrer)) {
-      return paywallConfig.referrer
+    if (paywallConfig.referrer && isEns(paywallConfig.referrer)) {
+      return true
     }
   }
-  return recipient
+  return false
+}
+
+export const getReferrers = async (
+  recipients: string[],
+  paywallConfig?: PaywallConfigType,
+  lockAddress?: string
+): Promise<string[]> => {
+  const isReferrerAddressEns = shouldReferrerResolveForENS(
+    paywallConfig,
+    lockAddress
+  )
+  if (isReferrerAddressEns) {
+    if (isReferrerAddressEns && paywallConfig && paywallConfig.referrer) {
```

Comment:
The reason to change this to be here is that if the referrer is an ENS it will be the same for all. There is no reason to call this for all recipients since that will unnecessarily make quite a few ENS calls.

Reply:
This codepath will be executed for any existing ENS addresses on config before this change goes live. since it will be formatted to address from here on out.

Reply:
I am not sure the referrer should ever be an ENS. You are resolving the ENS in the settings UI and the config only receives the actual address, which, IMO is good!